File: backend/assistants/utils/thought_integrity.py
from __future__ import annotations

def analyze_thought_integrity(content: str) -> str:
    if not content or len(content.strip()) < 10:
        return "empty"
    if "```json" in content and "{" not in content:
        return "markdown_stub"
    lower = content.lower()
    if "error" in lower or "traceback" in content:
        return "error_log"
    return "valid"

File: frontend/src/utils/diagnostics/thoughtIntegrity.js
export function analyzeThoughtIntegrity(content) {
  if (!content || content.trim().length < 10) return 'empty';
  if (content.includes('```json') && !content.includes('{')) return 'markdown_stub';
  if (content.toLowerCase().includes('error') || content.includes('traceback')) return 'error_log';
  return 'valid';
}
